app/controllers/api/v1/notes_controller.rb

```diff
@@ -0,0 +1,60 @@
+module Api
+  module V1
+    class NotesController < ApplicationController
+      def index
+        render_index_response
+      end
+
+      def show
+        render json: note, status: :ok, serializer: NoteSerializer
+      end
+
+      private
+
+      def note
+        Note.find(params.require(:id))
+      end
+
+      def render_index_response
+        if invalid_parameter?
+          render_error_message
+        else
+          render_ok
+        end
+      end
+
+      def render_error_message
+        if invalid_order?
+          render_invalid_parameter_error(I18n.t('invalid_order_error'))
+        elsif invalid_note_type?
+          render_invalid_parameter_error(I18n.t('invalid_note_type_error'))
+        end
+      end
```

Comment:
Está estructura quedó un poco compleja. Por un lado, recordá que la estructura de este tipo siempre es preferible ya que nos acorta y simplifica los métodos y los deja mucho más legibles.
```suggestion
      def render_index_response
        return render_error_message if invalid_parameter?

        render_ok
      end
```